tests/scripts/run_cloud_tests.py

```diff
@@ -1,19 +1,36 @@
 import os
+import signal
 import sys
-from create_test_suffix import set_index_suffix
+import time
+
+from create_and_set_cloud_unique_run_identifier import set_unique_run_identifier
 from delete_all_cloud_test_indexes import delete_all_test_indices
 from populate_indices_for_cloud_tests import populate_indices
 
+
+def handle_interrupt(signum, frame):
+    print("\nInterrupt received. Cleaning up and deleting indices.")
+    delete_all_test_indices()
+    sys.exit(1)
+
+
 if __name__ == '__main__':
-    # Generate the random suffix
-    set_index_suffix()
-    print(f"Using unique identifier: {os.environ.get('MQ_TEST_RUN_IDENTIFIER', '')}")
-    populate_indices()
-    print(f"All indices has been created, proceeding to run tests with pytest. Arguments: {sys.argv[1:]}")
+    # Set up the signal handler for KeyboardInterrupt (Cmd+C)
+    signal.signal(signal.SIGINT, handle_interrupt)
 
-    import pytest
-    pytest_args = ['tests/', '-m', 'not ignore_during_cloud_tests'] + sys.argv[1:]
-    pytest.main(pytest_args)
-    print("All tests has been executed, proceeding to delete indices")
+    test_start_time = time.time()
+    try:
+        set_unique_run_identifier()
+        print(f"Using unique identifier: {os.environ.get('MQ_TEST_RUN_IDENTIFIER', '')}")
```

Comment:
let's just fail if the env var is not set: `os.environ["MQ_TEST_RUN_IDENTIFIER"]`